source/common/http/header_utility.cc

```diff
@@ -117,6 +118,27 @@ bool HeaderUtility::matchHeaders(const Http::HeaderMap& request_headers,
   return match != header_data.invert_match_;
 }
 
+bool HeaderUtility::validateHeaders(const Http::HeaderMap& headers) {
+  bool headers_are_valid = true;
+
+  headers.iterate(
+      [](const Http::HeaderEntry& header, void* context) -> Http::HeaderMap::Iterate {
+        const absl::string_view header_value = header.value().getStringView();
+        if (nghttp2_check_header_value(
+                reinterpret_cast<const uint8_t*>(std::string(header_value).data()),
```

Comment:
Can you call .data on the string_view directly to avoid a temporary std::string?